octo-user-service/src/main/java/com/revature/octo/user/controller/BoardController.java

```diff
@@ -42,15 +51,95 @@
 		return new ResponseEntity<List<SystemUser>>(systemUserList, HttpStatus.OK);
 		
 	}
+	
+	@GetMapping(path="/getUsersOnBoard/{boardId}", produces=MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<List<SystemUser>> getUsersOnBoard(@PathVariable String boardId){
+		int boardNum = Integer.parseInt(boardId);
+		List<SystemUser> usersOnBoard = (List<SystemUser>)userRepo.findByBoardUserJoins_boardId(boardNum);
+		return new ResponseEntity<List<SystemUser>>(usersOnBoard, HttpStatus.OK);
+	}
+	
+	@PostMapping(path="/updateBoardUsers/{boardId}", consumes=MediaType.APPLICATION_JSON_VALUE, produces=MediaType.APPLICATION_JSON_VALUE)
+	@ResponseBody
+	public ResponseEntity<Boolean> updateBoardUsers(@PathVariable String boardId, @RequestBody List<SystemUser> updatedList){
+		//printing out the request body
+		System.out.println("\n\nupdated list of board users:" + updatedList + "\n\n");
+		
+		//boardId: String -> Integer
+		Integer boardNum = Integer.parseInt(boardId);
+		
+		//Get the list of board members prior to addition/deletion
+		List<SystemUser> currentBoardUsers = (List<SystemUser>) boardUserRepo.getSystemUsersByBoardId(boardNum);
+		
+		//user ids prior to addition/removal
+		List<Integer> oldUserIds = new ArrayList<Integer>();
+		for(SystemUser member : currentBoardUsers) {
+			oldUserIds.add(member.getId());
+		}
+		
+		//user ids after addition/removal 
+		List<Integer> newUserIds = new ArrayList<Integer>();
+		for(SystemUser newMember : updatedList) {
+			newUserIds.add(newMember.getId());
+		}
+		
+		SystemUser user = null;
+		BoardUserJoin buj = null;
+		
+		//DELETION:
+		//if the new list of board ids does NOT contain an id from before addition/deletion took place,
+		//delete BoardUserJoin from SystemUser, update boardUserRepo accordingly
+		for(Integer i : oldUserIds) {
+			if(!newUserIds.contains(i)) {
+				user = userRepo.findById(i);
+				System.out.println("\n\nUSERS BUJS BEFORE: " + user.getBoardUserJoins());
+				Set<BoardUserJoin> bujSet = user.getBoardUserJoins();
+				for(BoardUserJoin b : bujSet) {
+					if(b.getBoardId() == boardNum) {
+						bujSet.remove(b);
+						boardUserRepo.delete(b);
+						break;
+					}
+				}
+				System.out.println("\n\nUSERS BUJS AFTER: " + user.getBoardUserJoins() + "\n\n");
+				System.out.println("\n\nALL BUJS: " + boardUserRepo.findAll() + "\n\n");
+			}
+		}
+		
+		//ADDITION:
+		//Remove all previously existing board members from newUserIds list
+		//now the list will only contain new members' ids
+		newUserIds.removeAll(oldUserIds);
+		
+		//add BoardUserJoins per SystemUser, update boardUserRepo accordingly
+		for(Integer i : newUserIds) {
+			user = userRepo.findById(i);
+			buj = new BoardUserJoin(boardNum, user);
+			if(user.getBoardUserJoins() == null) {
+				HashSet<BoardUserJoin> bujs = new HashSet<BoardUserJoin>();
+				bujs.add(buj);
+				user.setBoardUserJoins(bujs);
+			}else {
+				user.getBoardUserJoins().add(buj);
+			}
+			boardUserRepo.save(buj);
```

Comment:
batch save/delete might be more responsive. also, it's probably better if there's an error that the error takes place for the whole batch instead of on an individual basis.